src/lib/Libifl/xml_encode_decode.c

```diff
@@ -512,10 +512,11 @@ extern	int decode_xml_arg_list_str(char *arg_list,
 
 	init_escapechars_maxarg(escape_chars, &arg_max);
 
-	/* Allocate memory to hold encoded argument */
-	arg = malloc(arg_max * sizeof(char *));
+	/* Allocate memory to hold decoded argument */
+	arg = malloc(strlen(arg_list) + 1);
```

Comment:
You call strlen() on arg_list twice.  Consider calling it only once and saving the result.